### ISSUE TITLE
Handling of breaks in try/finally leads to unsoundness

### ISSUE THREAD

**Issue by wmdietl** [typetools/checker-framework#548]
Take the following test case:

```
class TryFinallyBreak {
    String foo() {
        String ans = "x";
        while (true) {
            try {
                ans = "foo";
                break;
            }
            finally {
                ans = null;
            }
        }
        //:: error: (return.type.incompatible)
        return ans;
    }
}
```

The Nullness Checker should raise an error for the return statement, as the finally branch sets the value to null. However, it currently does not.
The code without 'break' works as expected.

Issue raised in "field initialization vs try block" discussion by Ed Price. Thanks!

Related to Issue #293 and #341 .

**Comment by wmdietl** [comment]
Added test case at:
checker/tests/initialization/fbc/TryFinallyBreak.java

**Comment by edp70** [comment]
Note: the test case can be minimized a bit more by omitting the assignment inside the try block (ans = "foo").

**Comment by mernst** [comment]
I have simplified the test case by removing the assignment inside the try block as suggested by @edp70.

**Comment by wmdietl** [comment]
Fixed by https://github.com/typetools/checker-framework/commit/04be2b98f5d19eab4b5288f55b0a3c600beffd90
